### PR TITLE
Propagate install_if and provider_priority to APKINDEX

### LINKED ISSUE #28704: Alpine package registry: Not all possible fields are present in generated APKINDEX.tar.gz
### Description

Hello,

in the current code for [generating the APKINDEX.tar.gz](https://github.com/go-gitea/gitea/blob/release/v1.21/services/packages/alpine/repository.go#L206) the following [supported fields](https://wiki.alpinelinux.org/wiki/Apk_spec#APKINDEX_Format) are not created and saved:

- k: - provider priority, integer (corresponds to provider_priority in PKGINFO), optional
- i: - install if (corresponds to install_if in PKGINFO, concatenated by spaces into a single line)

It would be nice if these fields can be added. I noticed the problem when installing the salt-minion package and the corresponding startup package (salt-minion-openrc) was not installed.

With regards
Joniw

### Gitea Version

1.21

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Test instance in Kubernetes

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by bugaevc**
Resolves https://github.com/go-gitea/gitea/issues/28704

Example of an entry in the generated `APKINDEX` file:

```
C:Q1xCO3H9LTTEbhKt9G1alSC87I56c=
P:hello
V:2.12-r1
A:x86_64
T:The GNU Hello program produces a familiar, friendly greeting
U:https://www.gnu.org/software/hello/
L:GPL-3.0-or-later
S:15403
I:36864
o:hello
m:
t:1705934118
D:so:libc.musl-x86_64.so.1
p:cmd:hello=2.12-r1
i:foobar=1.0 !baz
k:42
```

the `i:` and `k:` entries are new.

### CODE REVIEW

**Comment by bugaevc** [comment]
cc @joniw @mintyhippoxyz

**Review by KN4CK3R — CHANGES_REQUESTED** [review]
Thank you for the PR. This requires a migration because now the stored metadata is incompatible with the new struct.

**Comment by bugaevc** [comment]
Looks like `install_if` should be a single string and not an array after all: even though it has the same general format as `depends`/`provides`, `abuild` stores those as arrays (one per line), but formats `install_if` into a single space-separated line. So it makes sense to do just the same in the model. This also sidesteps the need for a migration.

PTAL

File: modules/packages/alpine/metadata.go [hunk 1]
```diff
@@ -188,6 +190,16 @@ func ParsePackageInfo(r io.Reader) (*Package, error) {
...
 			}
+		case "provider_priority":
+			n, err := strconv.ParseInt(value, 10, 64)
+			if err == nil {
+				p.FileMetadata.ProviderPriority = n
+			}
+		case "replaces_priority":
+			n, err := strconv.ParseInt(value, 10, 64)
+			if err == nil {
+				p.FileMetadata.ReplacesPriority = n
```
Comment (delvh) [inline modules/packages/alpine/metadata.go]: Did I miss something, or is the `ReplacesPriority` currently unused?

Comment (bugaevc) [reply modules/packages/alpine/metadata.go]: Yes, it is unused indeed. It is a part of Alpine package metadata, so it makes sense to parse and store it alongside all the other package metadata, but there isn't anything else being done with it currently. Specifically, it looks like there's no field for `replaces_priority` in the APKINDEX file, so `ReplacesPriority` is not used when generating APKINDEX.

I guess `ReplacesPriority` could show up in some HTTP API request if there is one that returns package file metadata, or if one is added in the future.

Comment (delvh) [reply modules/packages/alpine/metadata.go]: In that case, I recommend adding a comment to the field stating exactly that.

File: modules/packages/alpine/metadata.go [hunk 2]
```diff
@@ -55,16 +55,18 @@ type VersionMetadata struct {
...
+	BuildDate        int64    `json:"build_date,omitempty"`
+	Size             int64    `json:"size,omitempty"`
+	Architecture     string   `json:"architecture,omitempty"`
+	Origin           string   `json:"origin,omitempty"`
+	CommitHash       string   `json:"commit_hash,omitempty"`
+	InstallIf        string   `json:"install_if,omitempty"`
+	Provides         []string `json:"provides,omitempty"`
+	Dependencies     []string `json:"dependencies,omitempty"`
+	ProviderPriority int64    `json:"provider_priority,omitempty"`
+	ReplacesPriority int64    `json:"replaces_priority,omitempty"`
```
Comment (KN4CK3R) [inline modules/packages/alpine/metadata.go]:
```suggestion
```

File: modules/packages/alpine/metadata.go [hunk 3]
```diff
@@ -188,6 +190,16 @@ func ParsePackageInfo(r io.Reader) (*Package, error) {
...
+		case "provider_priority":
+			n, err := strconv.ParseInt(value, 10, 64)
+			if err == nil {
+				p.FileMetadata.ProviderPriority = n
+			}
+		case "replaces_priority":
+			n, err := strconv.ParseInt(value, 10, 64)
+			if err == nil {
+				p.FileMetadata.ReplacesPriority = n
+			}
```
Comment (KN4CK3R) [inline modules/packages/alpine/metadata.go]:
```suggestion
```

**Review by KN4CK3R — APPROVED** [review]
The migration is not needed, the field is just 0.

File: modules/packages/alpine/metadata.go (hunk 1 above)
Comment (KN4CK3R) [reply modules/packages/alpine/metadata.go]: I removed the field as it's not used in the index spec at the moment. We can add it if it is used in future.

https://wiki.alpinelinux.org/wiki/Apk_spec#APKINDEX_Format

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**